File: internal/parser/typescript/testdata/fixture/src/user.service.ts
import { User } from './user.entity';
import { UserRepository } from './user.repository';
import { Injectable } from '@nestjs/core';

@Injectable()
export class UserService {
    @Inject()
    private userRepository: UserRepository;

    getUser(id: number): User {
        return this.userRepository.findById(id);
    }

    loadUser(id: number): User {
        return this.getUser(id);
    }
}

File: internal/parser/java/testdata/fixture/src/main/java/com/example/demo/service/UserService.java
package com.example.demo.service;

import com.example.demo.model.User;
import com.example.demo.repository.UserRepository;

@Service
public class UserService {
    @Autowired
    private UserRepository userRepository;

    public User getUser(Long id) {
        return this.userRepository.findById(id);
    }

    public User loadUser(Long id) {
        return getUser(id);
    }
}
